### PR TITLE
use uint64_t instead of unsigned long long int

### PR DESCRIPTION
**Pull request by deepnum**
use <stdint.h> standard header file

### CODE REVIEW

File: registry/registry_internals.h [hunk 1]
```diff
@@ -12,28 +12,28 @@
 
 // ----------------------------------------------------------------------------
 // COMMON structures
-
+#include <stdint.h>
```
Comment (mfundul) [inline registry/registry_internals.h]: Please remove this line. It is included from `libnetdata.h` that already has:
```
#if defined(HAVE_INTTYPES_H)
#include <inttypes.h>
#elif defined(HAVE_STDINT_H)
#include <stdint.h>
#endif
```

**Comment by mfundul** [comment]
What is the motivation for this PR?

**Comment by deepnum** [comment]
https://matt.sh/howto-c

**Comment by deepnum** [comment]
why libnetdata.h has those lines

**Comment by mfundul** [comment]
> 
> 
> https://matt.sh/howto-c

This is plain wrong. There are plenty of use cases for types such as `char`, `long` etc, not to mention there are also a ton of libraries using those types that we actively use. Moreover netdata has to support various platforms that may or may not have `stdint.h`.

In your case it is probably better to use `uint64_t` since this looks like legacy code right there.

**Comment by mfundul** [comment]
> 
> 
> why libnetdata.h has those lines

Those lines are included everywhere and with the current structure you don't have to explicitly include any system library in other places. Those lines also cover various different platforms, that may or may not have the corresponding header file.

File: registry/registry_internals.h (hunk 1 above)
Comment (deepnum) [reply registry/registry_internals.h]: Ok

**Comment by deepnum** [comment]
>Moreover netdata has to support various platforms that may or may not have stdint.h.

@mfundul example?

**Comment by Ferroin** [comment]
> > Moreover netdata has to support various platforms that may or may not have stdint.h.
> 
> @mfundul example?

I'm rather curious myself.  The only possibility I can think of is Windows, but MSVC provides stdint.h on any of the versions we'd potentially be using (I mean, we're not seriously going to be building for XP are we?). Glibc, uClibc, Musl, Bionic, all three major BSD libc's, as well as Solaris, macOS, and pretty much every other UNIX platform that's been updated in the last decade has and provides it, so unless we're thinking of building Netdata in a standalone C environment (that is, like OS kernel code which has to provide everything itself), I'm not seeing any platforms we're considering supporting that don't have stdint.h.

**Comment by mfundul** [comment]
My phrasing was unfortunate, I meant prefering `inttypes.h` over `stdint.h` or not, as the code shows. Brief description in [wikipedia](https://en.wikipedia.org/wiki/C_data_types#inttypes.h).

**Pull request closed without merging**